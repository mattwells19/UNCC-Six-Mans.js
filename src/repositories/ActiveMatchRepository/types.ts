import { Team } from "../../types/common";

export interface PlayerInActiveMatch {
  id: number;
  team: Team;
  reportedTeam: Team | null;
  matchId: string;
}

export interface UpdatePlayerInActiveMatchInput {
  reportedTeam: Team;
}

export type NewActiveMatchInput = {
  id: number;
  team: Team;
};
