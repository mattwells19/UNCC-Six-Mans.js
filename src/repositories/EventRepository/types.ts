import { DateTime } from "luxon";

export interface Event {
  id: number;
  name: string;
  startDate: DateTime;
  endDate: DateTime | null;
  mmrMult: number;
}

export interface UpdateEventInput {
  name?: string;
  startDate?: DateTime;
  endDate?: DateTime | null;
  mmrMult?: number;
}
