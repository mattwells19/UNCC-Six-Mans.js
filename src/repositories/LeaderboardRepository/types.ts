import { BallChaser, Leaderboard } from "@prisma/client";
export interface PlayerStats {
  id: number;
  mmr: number;
  name: string;
  wins: number;
  losses: number;
  matchesPlayed: number;
  winPerc: number;
}

export interface UpdatePlayerStatsInput {
  id: number;
  mmr: number;
  wins?: number;
  losses?: number;
}

export type LeaderboardWithBallChaser = Leaderboard & { player: BallChaser };
