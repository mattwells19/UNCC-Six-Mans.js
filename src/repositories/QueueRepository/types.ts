import { BallChaser, Queue } from "@prisma/client";
import { DateTime } from "luxon";
import { Team } from "../../types/common";

export interface UpdateBallChaserInQueueInput {
  id: number;
  isCap?: boolean;
  team?: Team | null;
  queueTime?: DateTime;
}

export type PlayerInQueue = {
  id: number;
  isCap: boolean;
  mmr: number | null;
  name: string;
  queueTime: DateTime;
  team: Team | null;
};

export interface AddBallChaserToQueueInput {
  id: number;
  name: string;
  queueTime: DateTime;
}

export type QueueWithBallChaser = Queue & { player: BallChaser; team: Team | null };
