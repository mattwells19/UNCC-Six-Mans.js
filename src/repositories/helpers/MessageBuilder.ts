import { MessageActionRow, MessageButton, MessageEmbed } from "discord.js";
import { BallChaser } from "../../types/common";

export default class MessageBuilder {
  
  // eslint-disable-next-line max-len
  static readonly normIconURL = "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png";

  static readonly queueButtons = new MessageActionRow()
    .addComponents(
      new MessageButton()
        .setCustomId("joinQueue")
        .setLabel("Join")
        .setStyle("SUCCESS"),
      new MessageButton()
        .setCustomId("leaveQueue")
        .setLabel("Leave")
        .setStyle("DANGER"),
    );

  static leaderboardMessage(leaderboardInfo : string[]) : MessageEmbed[] {

    const embeds = leaderboardInfo.map((content, index) => {
      const embedCtr = leaderboardInfo.length > 1 ? `(${index + 1}/${leaderboardInfo.length})` : "";
  
      return new MessageEmbed()
        .setColor("BLUE")
        .setTitle(`UNCC 6 Mans | Leaderboard ${embedCtr}`.trim())
        .setThumbnail(this.normIconURL)
        .setDescription("```" + content + "```");
    });

    return embeds;
  }

  static activeQueueMessage( ballchasers : ReadonlyArray<Readonly<BallChaser>> ) : MessageEmbed {
    
    const ballChaserNames = ballchasers.map(function (a) { return a.name; }).join("\n");

    const embed = new MessageEmbed()
      .setColor("GREEN")
      .setTitle("Current Queue")
      .setThumbnail(this.normIconURL)
      .setDescription("Click the green button to join the queue!\n\n" +
                "Current Queue: " + ballchasers.length + "/6\n" + ballChaserNames);

    return embed;
  }

  static emptyQueueMessage() : MessageEmbed {

    const embed = new MessageEmbed()
      .setColor("GREEN")
      .setTitle("Queue is Empty")
      .setThumbnail(this.normIconURL)
      .setDescription("Click the green button to join the queue!");
    
    return embed;
  }
}