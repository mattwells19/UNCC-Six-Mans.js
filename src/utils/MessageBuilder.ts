import { MessageActionRow, MessageButton, MessageEmbed, MessageOptions } from "discord.js";
import { BallChaser } from "../types/common";

export const enum ButtonCustomID {
  JoinQueue = "joinQueue",
  LeaveQueue = "leaveQueue",
}

export default class MessageBuilder {
  private static readonly normIconURL =
    "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png";

  static leaderboardMessage(leaderboardInfo: string[]): MessageOptions {
    const embeds = leaderboardInfo.map((content, index) => {
      const embedCtr = leaderboardInfo.length > 1 ? `(${index + 1}/${leaderboardInfo.length})` : "";

      return new MessageEmbed({
        color: "BLUE",
        description: "```" + content + "```",
        thumbnail: { url: this.normIconURL },
        title: `UNCC 6 Mans | Leaderboard ${embedCtr}`.trim(),
      });
    });

    return {
      embeds,
    };
  }

  static disabledButtonsJoining(): MessageOptions {
    const joinButton = new MessageButton({
      customId: ButtonCustomID.JoinQueue,
      disabled: true,
      label: "Please wait...",
      style: "SUCCESS",
    });
    const leaveButton = new MessageButton({
      customId: ButtonCustomID.LeaveQueue,
      disabled: true,
      label: "Leave",
      style: "DANGER",
    });
    return {
      components: [new MessageActionRow({ components: [joinButton, leaveButton] })]
    };
  }

  static disabledButtonsLeaving(): MessageOptions {
    const joinButton = new MessageButton({
      customId: ButtonCustomID.JoinQueue,
      disabled: true,
      label: "Join",
      style: "SUCCESS",
    });
    const leaveButton = new MessageButton({
      customId: ButtonCustomID.LeaveQueue,
      disabled: true,
      label: "Please wait...",
      style: "DANGER",
    });
    return {
      components: [new MessageActionRow({ components: [joinButton, leaveButton] })]
    };
  }

  static enabledButtonsLeaving(): MessageOptions {
    const joinButton = new MessageButton({
      customId: ButtonCustomID.JoinQueue,
      label: "Join",
      style: "SUCCESS",
    });
    const leaveButton = new MessageButton({
      customId: ButtonCustomID.LeaveQueue,
      label: "Leave",
      style: "DANGER",
    });
    return {
      components: [new MessageActionRow({ components: [joinButton, leaveButton] })]
    };
  }

  static queueMessage(ballchasers: ReadonlyArray<Readonly<BallChaser>>): MessageOptions {
    const embed = new MessageEmbed({
      color: "GREEN",
      thumbnail: { url: this.normIconURL },
    });
    const joinButton = new MessageButton({
      customId: ButtonCustomID.JoinQueue,
      label: "Join",
      style: "SUCCESS",
    });
    const leaveButton = new MessageButton({
      customId: ButtonCustomID.LeaveQueue,
      label: "Leave",
      style: "DANGER",
    });

    if (ballchasers.length == 0) {
      embed.setTitle("Queue is Empty").setDescription("Click the green button to join the queue!");
    } else {
      const ballChaserList = ballchasers
        .map((ballChaser) => {
          // + 1 since it seems that joining the queue calculates to 59 instead of 60
          const queueTime = ballChaser.queueTime?.diffNow().as("minutes") ?? 0;
          return `${ballChaser.name} (${Math.min(queueTime + 1, 60).toFixed()} mins)`;
        })
        .join("\n");

      embed
        .setTitle(`Current Queue: ${ballchasers.length}/6`)
        .setDescription("Click the green button to join the queue!\n\n" + ballChaserList);
    }

    return {
      components: [new MessageActionRow({ components: [joinButton, leaveButton] })],
      embeds: [embed],
    };
  }
}
