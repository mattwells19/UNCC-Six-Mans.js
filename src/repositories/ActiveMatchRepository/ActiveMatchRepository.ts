import { ActiveMatch, Prisma, PrismaClient } from "@prisma/client";
import { PrismaClientKnownRequestError } from "@prisma/client/runtime";
import { Team } from "../../types/common";
import LeaderboardRepository from "../LeaderboardRepository";
import { generateRandomId, splitArray, waitForAllPromises } from "../../utils";
import { ActiveMatchTeams, NewActiveMatchInput, PlayerInActiveMatch, UpdatePlayerInActiveMatchInput } from "./types";

export class ActiveMatchRepository {
  #ActiveMatch: Prisma.ActiveMatchDelegate<Prisma.RejectOnNotFound | Prisma.RejectPerOperation | undefined>;

  constructor() {
    this.#ActiveMatch = new PrismaClient().activeMatch;
  }

  async addActiveMatch(newActiveMatchPlayers: Array<NewActiveMatchInput>): Promise<void> {
    const notEveryoneHasATeam = newActiveMatchPlayers.some((player) => !Number.isInteger(player.team));
    if (notEveryoneHasATeam) {
      throw new Error("Not all players are assigned a team.");
    }

    const matchId = generateRandomId();

    await this.#ActiveMatch.createMany({
      data: newActiveMatchPlayers.map((newActiveMatchPlayer) => ({
        id: matchId,
        playerId: newActiveMatchPlayer.id,
        team: newActiveMatchPlayer.team,
      })),
    });
  }

  async updatePlayerInActiveMatch(playerInMatchId: string, updates: UpdatePlayerInActiveMatchInput): Promise<void> {
    await this.#ActiveMatch
      .update({
        data: {
          brokenQueue: updates.brokenQueue,
          reportedTeam: updates.reportedTeam,
        },
        where: {
          playerId: playerInMatchId,
        },
      })
      .catch((err) => {
        if (err instanceof PrismaClientKnownRequestError && err.code === "P2025") {
          throw new Error(`Player with ID: ${playerInMatchId} is not in an active match.`);
        }
      });
  }

  async removeAllPlayersInActiveMatch(playerInMatchId: string): Promise<void> {
    await this.#ActiveMatch
      .findUnique({
        select: {
          id: true,
        },
        where: {
          playerId: playerInMatchId,
        },
      })
      .then((match) => {
        if (!match) {
          throw new Error(`Player with ID: ${playerInMatchId} is not in an active match.`);
        }

        return this.#ActiveMatch.deleteMany({
          where: {
            id: match.id,
          },
        });
      });
  }

  async getAllPlayersInActiveMatch(playerInMatchId: string): Promise<ActiveMatchTeams> {
    const allPlayersInMatch = await this.#ActiveMatch
      .findUnique({
        select: {
          id: true,
        },
        where: {
          playerId: playerInMatchId,
        },
      })
      .then((match) => {
        if (!match) {
          console.warn(`Player with ID: ${playerInMatchId} is not in an active match.`);
          return [];
        }

        return this.#ActiveMatch.findMany({
          where: {
            id: match.id,
          },
        });
      });

    const allPlayersInActiveMatch = await waitForAllPromises(allPlayersInMatch, async (playerInMatch) => {
      return await this.#getPlayerInActiveMatchWithMmr(playerInMatch);
    });

    const [blueTeam, orangeTeam] = splitArray(allPlayersInActiveMatch, (p) => p.team === Team.Blue);

    return {
      blueTeam,
      orangeTeam,
    };
  }

  async #getPlayerInActiveMatchWithMmr(playerInMatch: ActiveMatch): Promise<PlayerInActiveMatch> {
    const stats = await LeaderboardRepository.getPlayerStats(playerInMatch.playerId);
    return {
      id: playerInMatch.playerId,
      matchId: playerInMatch.id,
      mmr: stats ? stats.mmr : 100,
      reportedTeam: playerInMatch.reportedTeam,
      team: playerInMatch.team,
    };
  }

  async isPlayerInActiveMatch(playerInMatchId: string): Promise<boolean> {
    const playerInMatch = await this.#ActiveMatch.count({
      where: {
        playerId: playerInMatchId,
      },
    });

    return playerInMatch > 0;
  }

  async getPlayerInActiveMatch(playerInMatchId: string): Promise<PlayerInActiveMatch | null> {
    const playerInMatch = await this.#ActiveMatch.findUnique({
      where: {
        playerId: playerInMatchId,
      },
    });

    if (playerInMatch) {
      const player = this.#getPlayerInActiveMatchWithMmr(playerInMatch);
      return player;
    } else {
      return null;
    }
  }

  async isActiveMatch(): Promise<boolean> {
    const activeMatch = await this.#ActiveMatch.count();

    return activeMatch > 0;
  }
}

export default new ActiveMatchRepository();
