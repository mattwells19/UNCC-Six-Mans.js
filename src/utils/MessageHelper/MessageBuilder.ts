import {
  ButtonInteraction,
  EmbedField,
  MessageActionRow,
  MessageEmbed,
  MessageOptions,
  MessageSelectMenu,
  MessageSelectOptionData,
} from "discord.js";
import { ActiveMatchCreated } from "../../services/MatchService";
import { Team } from "../../types/common";
import { getEnvVariable } from "../utils";
import { PlayerInQueue } from "../../repositories/QueueRepository/types";
import EmbedBuilder from "./EmbedBuilder";
import ButtonBuilder from "./ButtonBuilder";
import CustomButton, { ButtonCustomID } from "./CustomButtons";

export const enum MenuCustomID {
  BlueSelect = "blueSelect",
  OrangeSelect = "orangeSelect",
}

export default class MessageBuilder {
  private static readonly isDev = getEnvVariable("ENVIRONMENT") === "dev";

  static leaderboardMessage(leaderboardInfo: string[]): MessageOptions {
    const embeds = leaderboardInfo.map((content, index) => {
      const embedCtr = leaderboardInfo.length > 1 ? `(${index + 1}/${leaderboardInfo.length})` : "";

      return EmbedBuilder.leaderboardEmbed("```" + content + "```", `UNCC 6 Mans | Leaderboard ${embedCtr}`.trim());
    });

    return {
      embeds,
    };
  }

  static queueMessage(ballchasers: ReadonlyArray<Readonly<PlayerInQueue>>): MessageOptions {
    let embed;
    if (ballchasers.length == 0) {
      embed = EmbedBuilder.queueEmbed("Queue is Empty", "Click the green button to join the queue!");
    } else {
      const ballChaserList = ballchasers
        .map((ballChaser) => {
          // + 1 since it seems that joining the queue calculates to 59 instead of 60
          const queueTime = ballChaser.queueTime?.diffNow().as("minutes") ?? 0;
          return `${ballChaser.name} (${Math.min(queueTime + 1, 60).toFixed()} mins)`;
        })
        .join("\n");

      embed = EmbedBuilder.queueEmbed(
        `Current Queue: ${ballchasers.length}/6`,
        "Click the green button to join the queue! \n\n" + ballChaserList
      );
    }

    return {
      components: [ButtonBuilder.queueButtons()],
      embeds: [embed],
    };
  }

  static fullQueueMessage(ballchasers: ReadonlyArray<Readonly<PlayerInQueue>>): MessageOptions {
    const ballChaserList = ballchasers
      .map((ballChaser) => {
        // + 1 since it seems that joining the queue calculates to 59 instead of 60
        const queueTime = ballChaser.queueTime?.diffNow().as("minutes") ?? 0;
        return `${ballChaser.name} (${Math.min(queueTime + 1, 60).toFixed()} mins)`;
      })
      .join("\n");

    const embed = EmbedBuilder.fullQueueEmbed("Click the Create Teams button to get started! \n\n" + ballChaserList);

    return {
      components: [ButtonBuilder.fullQueueButtons()],
      embeds: [embed],
    };
  }

  static activeMatchMessage({ blue, orange }: ActiveMatchCreated): MessageOptions {
    const embed = EmbedBuilder.activeMatchEmbed({ blue, orange });

    return {
      components: [ButtonBuilder.activeMatchButtons()],
      embeds: [embed],
    };
  }

  static captainChooseMessage(firstPick = true, ballChasers: ReadonlyArray<PlayerInQueue>): MessageOptions {
    //Get Available Players and Map players
    const availablePlayers: Array<MessageSelectOptionData> = [];
    const orangeTeam: Array<string> = [];
    const blueTeam: Array<string> = [];
    let captain = "";
    const embedColor = firstPick ? Team.Blue : Team.Orange;

    ballChasers.forEach((player) => {
      if (player.team === Team.Blue) {
        if (player.isCap && firstPick) {
          captain = player.name;
        }
        blueTeam.push("<@" + player.id + ">");
      } else if (player.team === Team.Orange) {
        if (player.isCap && !firstPick) {
          captain = player.name;
        }
        orangeTeam.push("<@" + player.id + ">");
      } else {
        availablePlayers.push({ description: "MMR: " + player.mmr, label: player.name, value: player.id });
      }
    });

    const playerChoices = new MessageSelectMenu();

    if (firstPick) {
      playerChoices.setCustomId(MenuCustomID.BlueSelect).setPlaceholder(captain + " choose a player");
    } else {
      playerChoices
        .setCustomId(MenuCustomID.OrangeSelect)
        .setPlaceholder(captain + " choose 2 players")
        .setMinValues(2)
        .setMaxValues(2);
    }

    playerChoices.addOptions(availablePlayers);

    const embed = EmbedBuilder.captainsChooseEmbed(embedColor, captain)
      .addField("ðŸ”· Blue Team ðŸ”·", blueTeam.join("\n"))
      .addField("ðŸ”¶ Orange Team ðŸ”¶", orangeTeam.join("\n"));

    const components = [new MessageActionRow({ components: [playerChoices] })];
    if (this.isDev) {
      components.push(ButtonBuilder.breakMatchButtons());
    }
    return {
      components,
      embeds: [embed],
    };
  }

  static reportedTeamButtons(buttonInteraction: ButtonInteraction, activeMatchEmbed: MessageEmbed): MessageOptions {
    let reportedTeam;
    const reportBlue = new CustomButton({ customId: ButtonCustomID.ReportBlue });
    const reportOrange = new CustomButton({ customId: ButtonCustomID.ReportOrange });

    switch (buttonInteraction.customId) {
      case ButtonCustomID.ReportBlue: {
        reportBlue.setStyle("PRIMARY");
        reportedTeam = "**Blue Team**";
        break;
      }
      case ButtonCustomID.ReportOrange: {
        reportOrange.setStyle("PRIMARY");
        reportedTeam = "**Orange Team**";
        break;
      }
    }
    const newField: EmbedField = {
      inline: false,
      name: "Reporting",
      value:
        "<@" +
        buttonInteraction.user +
        "> reported " +
        reportedTeam +
        " as the winner.\nAwaiting confirmation from the other team...\n" +
        "If this is incorrect, click the button of the correct team.",
    };
    const embed = new MessageEmbed(activeMatchEmbed);
    const updatedFields = embed.fields.map((field) => {
      if (field.name === "Reporting") {
        return newField;
      } else {
        return field;
      }
    });
    embed.setFields(updatedFields);

    return {
      components: [new MessageActionRow({ components: [reportBlue, reportOrange, ButtonBuilder.breakMatchButtons()] })],
      embeds: [embed],
    };
  }
}
