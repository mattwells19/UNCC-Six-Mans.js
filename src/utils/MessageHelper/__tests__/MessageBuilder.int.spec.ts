import faker from "faker";
import { BallChaserQueueBuilder } from "../../../../.jest/Builder";
import { ActiveMatchCreated } from "../../../services/MatchService";
import MessageBuilder from "../MessageBuilder";
import { PlayerInActiveMatch } from "../../../repositories/ActiveMatchRepository/types";
import { Team } from "../../../types/common";

jest.mock("../../utils");

describe("Building Buttons", () => {
  const mockBallChasers = BallChaserQueueBuilder.many(6);
  const mockMatchId = faker.datatype.uuid();

  it("return queue buttons", () => {
    const result = MessageBuilder.queueMessage(mockBallChasers);
    expect(result).toMatchSnapshot();
  });
  it("return full queue buttons", () => {
    const result = MessageBuilder.fullQueueMessage(mockBallChasers);
    expect(result).toMatchSnapshot();
  });
  it("return break match buttons", () => {
    const result = MessageBuilder.captainChooseMessage(true, mockBallChasers);
    expect(result).toMatchSnapshot();
  });
  it("return active match buttons", async () => {
    const orangePlayer: PlayerInActiveMatch = {
      id: mockBallChasers[0].id,
      team: Team.Orange,
      reportedTeam: null,
      matchId: mockMatchId,
      mmr: 100,
    };

    const bluePlayer: PlayerInActiveMatch = {
      id: mockBallChasers[1].id,
      team: Team.Blue,
      reportedTeam: null,
      matchId: mockMatchId,
      mmr: 100,
    };

    const activeMatch: ActiveMatchCreated = {
      blue: {
        mmrStake: 100,
        players: [bluePlayer],
        winProbability: 50,
      },
      orange: {
        mmrStake: 100,
        players: [orangePlayer],
        winProbability: 50,
      },
    };

    const result = MessageBuilder.activeMatchMessage(activeMatch);
    expect(result).toMatchSnapshot();
  });
});
