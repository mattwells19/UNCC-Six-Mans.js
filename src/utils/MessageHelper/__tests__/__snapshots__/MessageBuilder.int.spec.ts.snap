// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building Buttons return active match buttons 1`] = `
Object {
  "components": Array [
    Object {
      "components": Array [
        Object {
          "custom_id": "reportBlue",
          "disabled": false,
          "emoji": null,
          "label": "ðŸ”· Blue Team Won ðŸ”·",
          "style": 2,
          "type": 2,
          "url": null,
        },
        Object {
          "custom_id": "reportOrange",
          "disabled": false,
          "emoji": null,
          "label": "ðŸ”¶ Orange Team Won ðŸ”¶",
          "style": 2,
          "type": 2,
          "url": null,
        },
      ],
      "type": 1,
    },
  ],
  "embeds": Array [
    Object {
      "author": null,
      "color": 10038562,
      "description": null,
      "fields": Array [
        Object {
          "inline": false,
          "name": "ðŸ”· Blue Team ðŸ”·",
          "value": "<@e4e5133a-c1c4-44fd-aafd-35bb8ae68b93>",
        },
        Object {
          "inline": false,
          "name": "ðŸ”¶ Orange Team ðŸ”¶",
          "value": "<@8c5f7de0-a293-44d4-8ddb-36cacdecf4c5>",
        },
        Object {
          "inline": false,
          "name": "MMR Stake & Probability Rating:
",
          "value": "ðŸ”· Blue Team: Â Â Â Â Â Â Â Â **(+100)**Â Â **(-100)** ðŸ”·
ðŸ”¶ Orange Team:Â Â **(+100)**Â Â **(-100)** ðŸ”¶
Both teams are predicted to have a **50%** chance of winning.",
        },
      ],
      "footer": null,
      "image": null,
      "thumbnail": Object {
        "height": undefined,
        "proxyURL": undefined,
        "url": "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png",
        "width": undefined,
      },
      "timestamp": null,
      "title": "Teams are set!",
      "type": "rich",
      "url": null,
    },
  ],
}
`;

exports[`Building Buttons return break match buttons 1`] = `
Object {
  "components": Array [
    Object {
      "components": Array [
        Object {
          "custom_id": "blueSelect",
          "disabled": false,
          "max_values": undefined,
          "min_values": null,
          "options": Array [],
          "placeholder": "partnerships choose a player",
          "type": 3,
        },
      ],
      "type": 1,
    },
  ],
  "embeds": Array [
    Object {
      "author": null,
      "color": 3447003,
      "description": "ðŸ”· partnerships ðŸ”· chooses first",
      "fields": Array [
        Object {
          "inline": false,
          "name": "ðŸ”· Blue Team ðŸ”·",
          "value": "<@8c5f7de0-a293-44d4-8ddb-36cacdecf4c5>",
        },
        Object {
          "inline": false,
          "name": "ðŸ”¶ Orange Team ðŸ”¶",
          "value": "<@e4e5133a-c1c4-44fd-aafd-35bb8ae68b93>
<@4ff408bc-b824-4018-b766-a7d8c3134535>
<@5bbc975a-75d0-43e5-adc8-cdeec16f86e8>
<@5671e947-8a0d-4a06-b00d-1ec287b84315>
<@5b438291-c9df-47b2-8e94-17902ae7328c>",
        },
      ],
      "footer": null,
      "image": null,
      "thumbnail": Object {
        "height": undefined,
        "proxyURL": undefined,
        "url": "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png",
        "width": undefined,
      },
      "timestamp": null,
      "title": null,
      "type": "rich",
      "url": null,
    },
  ],
}
`;

exports[`Building Buttons return full queue buttons 1`] = `
Object {
  "components": Array [
    Object {
      "components": Array [
        Object {
          "custom_id": "chooseTeam",
          "disabled": false,
          "emoji": null,
          "label": "Captains",
          "style": 1,
          "type": 2,
          "url": null,
        },
        Object {
          "custom_id": "randomizeTeams",
          "disabled": false,
          "emoji": null,
          "label": "Random",
          "style": 1,
          "type": 2,
          "url": null,
        },
        Object {
          "custom_id": "leaveQueue",
          "disabled": false,
          "emoji": null,
          "label": "Leave",
          "style": 4,
          "type": 2,
          "url": null,
        },
      ],
      "type": 1,
    },
  ],
  "embeds": Array [
    Object {
      "author": null,
      "color": 5763719,
      "description": "Click the Create Teams button to get started! 

partnerships (60 mins)
Small (60 mins)
Savings (60 mins)
Checking (60 mins)
invoice (60 mins)
Granite (60 mins)",
      "fields": Array [],
      "footer": null,
      "image": null,
      "thumbnail": Object {
        "height": undefined,
        "proxyURL": undefined,
        "url": "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png",
        "width": undefined,
      },
      "timestamp": null,
      "title": "Queue is Full",
      "type": "rich",
      "url": null,
    },
  ],
}
`;

exports[`Building Buttons return queue buttons 1`] = `
Object {
  "components": Array [
    Object {
      "components": Array [
        Object {
          "custom_id": "joinQueue",
          "disabled": false,
          "emoji": null,
          "label": "Join",
          "style": 3,
          "type": 2,
          "url": null,
        },
        Object {
          "custom_id": "leaveQueue",
          "disabled": false,
          "emoji": null,
          "label": "Leave",
          "style": 4,
          "type": 2,
          "url": null,
        },
      ],
      "type": 1,
    },
  ],
  "embeds": Array [
    Object {
      "author": null,
      "color": 5763719,
      "description": "Click the green button to join the queue! 

partnerships (60 mins)
Small (60 mins)
Savings (60 mins)
Checking (60 mins)
invoice (60 mins)
Granite (60 mins)",
      "fields": Array [],
      "footer": null,
      "image": null,
      "thumbnail": Object {
        "height": undefined,
        "proxyURL": undefined,
        "url": "https://raw.githubusercontent.com/mattwells19/UNCC-Six-Mans.js/main/media/norm_still.png",
        "width": undefined,
      },
      "timestamp": null,
      "title": "Current Queue: 6/6",
      "type": "rich",
      "url": null,
    },
  ],
}
`;
